bin/cp/utils.c

```diff
@@ -112,6 +112,8 @@ copy_file(const FTSENT *entp, int dne)
 		if ((from_fd = open(entp->fts_path, O_RDONLY, 0)) < 0 ||
 		    fstat(from_fd, &sb) != 0) {
 			warn("%s", entp->fts_path);
+			if (from_fd >= 0)
```

Comment:
You could just close it since you're casting away the return value anyway.

Reply:
I'd rather avoid calling close with an invalid fd, and am happy with the change as proposed

Reply:
OK